File: src/GoatSlips/Models/Database/Project.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace GoatSlips.Models.Database
{
    public sealed class Project
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public decimal Rate { get; set; }
        public string? FirstName { get; set; }
        public string? LastName { get; set; }
        public string? BusinessName { get; set; }
        public string? Email { get; set; }
        public string? Address1 { get; set; }
        public string? Address2 { get; set; }
        public string? City { get; set; }
        public string? State { get; set; }
        public int? Zip { get; set; }
        public int? ZipExtension { get; set; }
        public DateTime? LockDate { get; set; }
    }
}

File: src/GoatSlips/Client/src/types/Project.ts
import { DropdownOption } from './DropdownOption';

export interface Project extends DropdownOption {
    rate: number;
    firstName: string;
    lastName: string;
    businessName: string;
    email: string;
    address1: string;
    address2: string;
    city: string;
    state: string;
    zip: number | null;
    zipExtension: number | null;
    lockDate: Date | null;
}
